graalpython/com.oracle.graal.python/src/com/oracle/graal/python/builtins/modules/BuiltinFunctions.java

```diff
@@ -286,6 +294,209 @@ public Object absObject(VirtualFrame frame, Object object,
         }
     }
 
+    /**
+     * Common class for all() and any() operations, as their logic and behaviors are very similar.
+     */
+    public abstract static class AllOrAnyNode extends PNodeWithContext {
+        enum NodeType {
+            ALL,
+            ANY
+        }
+
+        @Child private PyObjectIsTrueNode isTrueNode = PyObjectIsTrueNode.create();
+
+        abstract boolean execute(Frame frame, Object storageObj, NodeType nodeType);
+
+        @Specialization
+        boolean doBoolSequence(VirtualFrame frame,
+                        BoolSequenceStorage sequenceStorage,
+                        NodeType nodeType) {
+            boolean[] internalArray = sequenceStorage.getInternalBoolArray();
+
+            for (int i = 0; i < sequenceStorage.length(); i++) {
```

Comment:
Optional: The loop condition (of all such `for` loops) could be profiled using `LoopConditionProfile`.

Reply:
Can an instance of `LoopConditionProfile` be kept in a field, or is it better for every specialization method to take one as parameter and feed it to the `AllOrAnyNode` methods?